dev-guide.adoc

```diff
@@ -0,0 +1,226 @@
+= rh-che dev guide
+:toc:
+:toc-title:
+
+== Building rh-che
+With the exception of the dashboard component, rh-che is a straightforward maven project and can be built as expected. For additional information on the build process for the dashboard component, see link:assembly/fabric8-ide-dashboard-war/README.md[README.md] in the dashboard subcomponent. 
+
+[NOTE]
+====
+The rh-che maven build accomodates the same parameters as the upstream maven build
+====
+
+==== Basic build of all components:
+[source,bash]
+----
+mvn clean install
+----
+
+==== Build only only one component and dependents (e.g. dashboard component):
+[source,bash]
+----
+mvn clean install -pl ':fabric8-ide-dashboard-war' --amd 
+----
+
+==== Build, skipping tests and optional checks
+Use the `-Pfast` option:
+[source,bash]
+----
+mvn clean install -Pfast
+----
+This is equivalent to 
+[source,bash]
+----
+mvn -DskipTests=true \
+    -Dfindbugs.skip=true \
+    -Dmdep.analyze.skip=true \
+    -Dlicense.skip=true \
+    -Dgwt.compiler.localWorkers=2 -T 1C
+----
+Which can be slightly sped up further by adding `-Dskip-validate-sources -Dskip-enforce`.
+
+==== Add license headers to all files (fix license check step of full build):
+[source,bash]
+----
+mvn com.mycila:license-maven-plugin:format
+----
+
+==== Format all files (fix check-format step of full build):
+[source,bash]
+----
+mvn fmt:format
+----
+
+== Creating docker images
+[NOTE]
+====
+The scripts in this section assume that the maven build was completed successfully and that built artifacts exist at `+./assembly/assembly-main/target/eclipse-che-*/eclipse-che-*+`
```

Comment:
We can copy files using Dockerfile and .dockerignore files. Shouldn't we use this technique instead of scripts?

Reply:
I've seen some issues around build contexts and docker version, with inconsistent support. I agree that it would be better but with the availability of docker versions I sure wouldn't want to test it :)

Reply:
For clarity, these scripts are simplifications of how the cico build happens right now, so it would be worth changing it in both cases.

Reply:
I think you are referring to the multi-stage build which is a pretty new feature. Whereas .dockerignore is ancient (if not thinking of how long docker exists at all) and should be supported everywhere I believe. But I'm OK with the current improvements, just trying to simplify everything